prepare_check_cleanup/migration.py

```diff
@@ -0,0 +1,269 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Utility for running the migration action, primarily for testing purposes."""
+
+import argparse
+import json
+import logging
+import os
+import sys
+from contextlib import suppress
+from enum import Enum
+from pathlib import Path
+
+from github.GithubException import GithubException
+from github.GitRef import GitRef
+from github.PullRequest import PullRequest
+from github.Repository import Repository
+
+from prepare_check_cleanup import exit_
+from src.discourse import create_discourse
+from src.exceptions import DiscourseError
+from src.index import DOCUMENTATION_FOLDER_NAME, DOCUMENTATION_INDEX_FILENAME
+from src.pull_request import (
+    ACTIONS_PULL_REQUEST_TITLE,
+    DEFAULT_BRANCH_NAME,
+    create_repository_client,
+)
+from src.reconcile import NAVIGATION_TABLE_START
+
+
+class Action(str, Enum):
+    """The actions the utility can take.
+
+    Attrs:
+        PREPARE: Prepare discourse pages before running the migration.
+        CHECK_BRANCH: Check that the migration branch was created.
+        CHECK_PULL_REQUEST: Check that the migration pull request was created.
+        CLEANUP: Delete discourse pages before after the migration.
```

Comment:
`before after` ?
Should probably rephrase :)

Reply:
Yes that is confusing, should be fixed shortly.